### PR TITLE
add 'Activate Parent Environment' command

### PR DESCRIPTION
**Pull request by aviatesk**
This is especially useful when invoked from active editor, etc.

### CODE REVIEW

File: scripts/packages/VSCodeServer/src/misc.jl [hunk 1]
```diff
@@ -139,3 +139,24 @@ function activate_uri(conn, uri)
...
+end
+
+function search_up_file(basename, dir)
+    parent_dir = dirname(dir)
+    return if (parent_dir == dir || # ensure to escape infinite recursion
+        isempty(dir)) # reached to the system root
+        nothing
+    else
+        path = joinpath(dir, basename)
+        isfile(path) ? path : search_up_file(basename, parent_dir)
```
Comment (pfitzseb) [inline scripts/packages/VSCodeServer/src/misc.jl:160]: Probably better to use a "safe" `isfile` (e.g. the usual `isfile′(path) = try isfile(path) catch err false end`).

File: scripts/packages/VSCodeServer/src/misc.jl [hunk 2]
```diff
@@ -139,3 +139,24 @@ function activate_uri(conn, uri)
     Pkg.activate(uri)
     return nothing
 end
+
+function activate_from_dir(conn, dir)
+    path = search_up_file("Project.toml", dir)
+    if path === nothing
+        @warn "No Project.toml found for `$(dir)`"
```
Comment (pfitzseb) [inline scripts/packages/VSCodeServer/src/misc.jl:146]: I think I'd prefer to show this in the VSCode UI. Maybe return a `(error = true, dir = dir)` NT here instead and show a notification in VSCode?

**Review by pfitzseb — CHANGES_REQUESTED** [review]

File: scripts/packages/VSCodeServer/src/misc.jl (hunk 1 above)
Comment (aviatesk) [reply scripts/packages/VSCodeServer/src/misc.jl]: yeah, good point

File: scripts/packages/VSCodeServer/src/misc.jl (hunk 2 above)
Comment (aviatesk) [reply scripts/packages/VSCodeServer/src/misc.jl]: addressed

**Review by pfitzseb — APPROVED** [review]

**Pull request merged**